lib/assert.js

```diff
@@ -330,7 +330,14 @@ function _throws(shouldThrow, block, expected, message) {
     fail(actual, expected, 'Missing expected exception' + message);
   }
 
-  if (!shouldThrow && expectedException(actual, expected)) {
+  const userProvidedMessage = typeof message === 'string';
```

Comment:
Just because I stumbled upon this comment right now: this is fixed in newer versions.